library/Zend/Db/Sql/AbstractSql.php

```diff
@@ -108,6 +108,13 @@ protected function processExpression(ExpressionInterface $expression, PlatformIn
         $expressionParamIndex = &$this->instanceParameterIndex[$namedParameterPrefix];
 
         foreach ($parts as $part) {
+            // #7407: use $expression->getExpression() to get the unescaped version of the expression
+            if (is_string($part) && $expression instanceof Expression) {
```

Comment:
May be it'll be better:

    if (is_string($part)) {
        if ($expression instanceof Expression) {
            // #7407: use $expression->getExpression() to get the unescaped version of the expression
            $sql .= $expression->getExpression();
        } else {
            // simply tack it onto the return sql "specification" string
            $sql .= $part;
        }

        continue;
    } else (is_array($part)) {
        //
    } else {
        throw // Elements returned from getExpressionData() array must be a string or array.
    }